src/python/pants/backend/scala/subsystems/scalac.py

```diff
@@ -30,7 +30,7 @@ class Scalac(Subsystem):
             "set `[scalac].plugins_global_lockfile`."
         ),
         advanced=True,
-        removal_version="2.12.0.dev2",
+        removal_version="2.12.0.dev3",
```

Comment:
@tdyas and @chrisjrn I gave this the old college try and came up short. Although I got everything compiling and seemingly running, I could find no way to get scala_test.py fully working. I was hack-a-mole / whac-a-mole not fully understanding where global plugins should come from. Seemed like they should come from the "scalac-plugins" jvm resolve, but I was floundering so I punted.

Reply:
There are no global plugins any more once those options come out. They would be resolve-scoped using `[scalac].plugins_for_resolve`

Reply:
Ok, perhaps you or Chris could actually take up doing the deprecation before 2.12.0.dev3. I'm happy to review to see what I missed trying to get this working.